File: src/main/java-graphql/io/github/graphql/model/StargazerEdgeTO.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer;
import java.util.StringJoiner;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:47+0800"
)
public class StargazerEdgeTO implements java.io.Serializable {

    @javax.validation.constraints.NotNull
    private String cursor;
    @javax.validation.constraints.NotNull
    private UserTO node;
    @javax.validation.constraints.NotNull
    private String starredAt;

    public StargazerEdgeTO() {
    }

    public StargazerEdgeTO(String cursor, UserTO node, String starredAt) {
        this.cursor = cursor;
        this.node = node;
        this.starredAt = starredAt;
    }

    public String getCursor() {
        return cursor;
    }

    public UserTO getNode() {
        return node;
    }

    public String getStarredAt() {
        return starredAt;
    }


    @Override
    public String toString() {
        StringJoiner joiner = new StringJoiner(", ", "{ ", " }");
        if (cursor != null) {
            joiner.add("cursor: " + GraphQLRequestSerializer.getEntry(cursor));
        }
        if (node != null) {
            joiner.add("node: " + GraphQLRequestSerializer.getEntry(node));
        }
        if (starredAt != null) {
            joiner.add("starredAt: " + GraphQLRequestSerializer.getEntry(starredAt));
        }
        return joiner.toString();
    }

    public static StargazerEdgeTO.Builder builder() {
        return new StargazerEdgeTO.Builder();
    }

    public static class Builder {

        private String cursor;
        private UserTO node;
        private String starredAt;

        public Builder() {
        }

        public Builder setCursor(String cursor) {
            this.cursor = cursor;
            return this;
        }

        public Builder setNode(UserTO node) {
            this.node = node;
            return this;
        }

        public Builder setStarredAt(String starredAt) {
            this.starredAt = starredAt;
            return this;
        }


        public StargazerEdgeTO build() {
            return new StargazerEdgeTO(cursor, node, starredAt);
        }

    }
}

File: src/main/scala-graphql/io/github/graphql/model/StargazerEdgeTO.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:00+0800"
)
class StargazerEdgeTO(
    @javax.validation.constraints.NotNull
    cursor: String,
    @javax.validation.constraints.NotNull
    node: UserTO,
    @javax.validation.constraints.NotNull
    starredAt: String
) {

    override def toString(): String = {
        Seq(
            if (cursor != null) "cursor: " + GraphQLRequestSerializer.getEntry(cursor) else "",
            if (node != null) "node: " + GraphQLRequestSerializer.getEntry(node) else "",
            if (starredAt != null) "starredAt: " + GraphQLRequestSerializer.getEntry(starredAt) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

object StargazerEdgeTO {

    def builder(): StargazerEdgeTO.Builder = new Builder()

    class Builder {

        private var cursor: String = _
        private var node: UserTO = _
        private var starredAt: String = _

        def setCursor(cursor: String): Builder = {
            this.cursor = cursor
            this
        }

        def setNode(node: UserTO): Builder = {
            this.node = node
            this
        }

        def setStarredAt(starredAt: String): Builder = {
            this.starredAt = starredAt
            this
        }

        def build(): StargazerEdgeTO = new StargazerEdgeTO(cursor, node, starredAt)

    }
}
